src/language_features/code_lens.rs

```diff
@@ -32,10 +32,54 @@ pub fn text_document_code_lens(meta: EditorMeta, ctx: &mut Context) {
         partial_result_params: PartialResultParams::default(),
     };
     ctx.call::<CodeLensRequest, _>(meta, req_params, |ctx: &mut Context, meta, result| {
-        editor_code_lens(meta, result, ctx)
+        editor_code_lens(meta.clone(), result.clone(), ctx);
+        inlay_hints_from_editor_code_lens(meta, result.unwrap_or_default(), ctx)
     });
 }
 
+fn inlay_hints_from_editor_code_lens(
```

Comment:
Great idea. I forgot to document why I didn't do this yet: it's just because I hadn't used a language server where always-visible code lenses are useful.

(optional) I wonder if this should be enabled by default. I'd definitely vote for that if replace-ranges was less buggy.

Reply:
Editors like vscode I think choose to show all the code-lens always. 

However due to https://github.com/mawww/kakoune/issues/4659 the experience in kakoune is not always ideal. However, in-spite of that, I find it useful to keep this on for my own editing always. If it were not for that issue, I would definitely show these always. 

Anyways I leave it to you to take a call!